File: coefficiente binomiale/cb.py



# Formula
# risultato = n! /  k! (n-k)!

# funzione fattoriale (ricorsiva)
# 5! = 5*4*3*2*1

# Quando ti sei fermato? 1 => caso base
# A ogni passaggio cosa hai fatto? => n * (n-1) => passo ind.
def fattoriale(n):
  if (n==1):
    return 1
  else:
    return n*fattoriale(n-1)

# funzione coeff bin 
def coef (n,k):
  risultato=fattoriale(n)/(fattoriale(k)*fattoriale(n-k))
  return risultato
    
# __main__

val1 = int(input("Inserisci il primo numero: "))
val2 = int(input("Inserisci il secondo numero: "))

ris = coef(val1, val2)

print("Il risultato e': " + str(ris))

File: coefficiente binomiale/cb.c
#include <stdio.h>

// Funzione fattoriale che calcola il fattoriale di un numero
int fattoriale(int n){
  if (n == 1){
    return 1;
  } else {
    return n * fattoriale(n - 1);
  }
}

// Funzione coefBin che calcola il coefficiente binomiale
int coefBin(int n, int k){
  int risultato = fattoriale(n) / (fattoriale(k) * fattoriale(n-k));
  return risultato;
}

// Funzione main
int main(){
  int val1, val2, ris;

  printf("Inserisci il primo numero: ");
  scanf("%d", &val1);

  printf("Inserisci il secondo numero: ");
  scanf("%d", &val2);

  ris = coefBin(val1, val2);

  printf("Il risultato e': %d\n", ris);

  return 0;
}
